source/elements/oneDPL/source/parallel_api/buffer_wrappers.rst

```diff
@@ -43,33 +43,50 @@ Buffer Wrappers
 
 ``oneapi::dpl::begin`` and ``oneapi::dpl::end`` are helper functions
 for passing `SYCL`_ buffers to oneDPL algorithms.
-These functions accept a buffer and return an object
-of an unspecified type that satisfies the following requirements:
-
-- it is ``CopyConstructible``, ``CopyAssignable``, and comparable
-  with operators ``==`` and ``!=``;
-- the following expressions are valid: ``a + n``, ``a - n``,
-  ``a - b``, where ``a`` and ``b`` are objects of the type,
-  and ``n`` is an integer value;
-- it provides the ``get_buffer()`` method that returns the buffer
-  passed to the ``begin`` or ``end`` function.
+These functions accept a SYCL buffer and return an object of an unspecified type
+(referred below as *buffer position*) that satisfies the following requirements:
+
+- It is ``CopyConstructible`` and ``CopyAssignable``.
+- It is comparable with operators ``==`` and ``!=``.
+- It provides the ``get_buffer()`` method that returns the SYCL buffer, over which an object was built.
+- The expressions ``a + n`` and ``a - n``, where ``a`` is a buffer position object and ``n``
+  is an integer value, are valid and evaluate to a buffer position object representing
+  a position in the buffer that follows or precedes the position of ``a`` by ``n``.
+  If this new position is out of the buffer bounds, the behavior is undefined.
+- The expression ``a - b``, where ``a`` and ``b`` are buffer position objects,
+  is valid and evaluates to an integer value ``n`` such that ``a == b + n``.
+
+These operations are only valid for objects built over the same SYCL buffer.
+If an expression operates with buffer position objects that are not built
+over the same buffer, the behavior is undefined.
 
 When invoking an algorithm, the buffer passed to ``begin`` should be the same
 as the buffer passed to ``end``. Otherwise, the behavior is undefined.
 
+An algorithm may return a buffer position object, which can then be compared and used
+in expressions with other objects built over the same buffer.
+
```

Comment:
Would it be helpful to explicitly mention that there is **no** requirement that these types are dereferenceable on the host, and there is **no** requirement that a buffer index is retrievable from these objects?

In the documentation, I think it would be useful to call this out, in a specification, I'm not sure.  
That is clearly left unspecified already, but explicit mention may give more clarity about the intentions for how these buffer position objects are to be used (and perhaps more importantly how they are not intended to be used).

Reply:
Added a note to clarify the intent.